astropy/table/table.py

```diff
@@ -896,24 +900,30 @@ def show_in_notebook(self, tableid=None, css=None, display_length=50,
                                             np.random.randint(1, 1e6))
 
         jsv = JSViewer(display_length=display_length)
-        html = self._base_repr_(html=True, max_width=-1, tableid=tableid,
-                                max_lines=-1, show_dtype=False,
-                                tableclass=table_class)
+
+        has_row_index = 'row_index' in self.colnames
+        try:
```

Comment:
I might suggest creating a temporary table.  This is cheap because of the `copy=False`, and maybe even faster in the end than adding a column and then removing it:
```
row_col_name = 'row'
while row_col_name in self.columns:
    row_col_name = ' ' + row_col_name

row_col = self.ColumnClass(name=row_col_name, data=np.arange(len(self)))
display_table = self.__class__([row_col] + self.columns.values(), copy=False)
html = display_table._base_repr(html=True, ...)
```